infra/src/main/kotlin/fi/fta/geoviite/infra/geometry/GeometryService.kt

```diff
@@ -343,6 +356,36 @@ class GeometryService @Autowired constructor(
         return FileName("$VERTICAL_GEOMETRY ${locationTrack.name}") to csvFileContent.toByteArray()
     }
 
+    @Scheduled(
+        cron = "\${geoviite.rail-network-export.vertical-geometry-schedule}"
+    )
+    fun makeEntireVerticalGeometryListingCsv() = runVerticalGeometryListGeneration {
+        logger.serviceCall("makeVerticalGeometryListingCsv")
+        val trackNumberAndGeocodingContextCache = trackNumberService.listOfficial().associate { tn ->
+            tn.id to (tn to geocodingService.getGeocodingContext(OFFICIAL, tn.id))
+        }
+        val verticalGeometryListingWithTrackNumbers = locationTrackService.list(OFFICIAL, includeDeleted = false)
+            .sortedBy { locationTrack -> locationTrack.name }
```

Comment:
Tekeekö tämä sortedBy mitään?

Reply:
Muutin tämän intuitiivisempaan muotoon.

Tuohan siis itseasissa kyllä järjesti listan oikein aiemminkin, mutta ei kyllä ollut nopeasti luettuna ollenkaan selvää miksi: Kun listan järjestää ensin toissijaisen järjestyksen perusteella ja uudelleen ensisijaisen järjestyksen perusteella, lopputulos on oikein, JOS käytetään stabiilia järjestyalgoritmia. Kotlinin järjestysfunktiot on siis speksattu stabiileiksi ("equal elements preserve their order relative to each other after sorting").

Reply:
huomattavasti selkeempi 👍